### DISCUSSION TITLE
Will Warp-iMagick generate jpg fallback alternatives from WebP original?

### DISCUSSION THREAD

**Discussion by com2** [ddur/Warp-iMagick#3] [Ideas]
Imagine a seller who wants to support the poor among his prospective clients that still use Safari/OS without WebP support. Imagine that  the seller's uploaded images are already in that format? Will Warp-iMagick generate and serve the alternative fallback jpg graphics to browsers without WebP support?

**Comment by ddur** [comment]
@com2 

> Imagine a seller who wants to support the poor among his prospective clients that still use Safari/OS without WebP support. Imagine that the seller's uploaded images are already in that format? Will Warp-iMagick generate and serve the alternative fallback jpg graphics to browsers without WebP support?

Currently no, plugin is activated only on JPEG/PNG, but that is a good idea to consider implementing. 
Thanks.

**Reply by ddur** [reply]
@Magnifin 

> I apologize for the offtopic. But I really liked your plugin. I am not a programmer. But on the Internet, I found a very useful video-flv-converter plugin and modified it to fit my needs. I think the author will be interested in making a version for everyone. I'll post the modified code in the next post. Stand with Ukraine

Warp iMagick plugin handles only JPEG/PNG images. That plugin may handle videos without conflict with Warp iMagick. But that plugin requires system binaries to work, using [PHP exec](https://www.php.net/manual/en/function.exec.php) function which is usually forbidden on most of hosting providers. Warp iMagick is made by design to be safe, to use PHP modules, to work on all hosting providers (where [PHP ImageMagick](https://www.php.net/manual/en/book.imagick.php) & [PHP-GD](https://www.php.net/manual/en/book.image.php) modules are available) and never require executing system (or other) binaries.

**Comment by ddur** [comment]
@com2 

> Imagine a seller who wants to support the poor among his prospective clients that still use Safari/OS without WebP support. Imagine that the seller's uploaded images are already in that format? Will Warp-iMagick generate and serve the alternative fallback jpg graphics to browsers without WebP support?

However, almost all images available from web stock photos, from photo software and of course from digital cameras are in JPEG format.

So you can just skip JPEG to WebP conversion step, upload JPEG and have your conversion automagically done.

And have same fallback problem solved.

As serving WebP clone images is done transparently by web server software, there is no redirection (301/302) delay.

BTW, If your page/post HTML img URL is pointing to WebP clone image URL, that image will be directly served.

**Comment by ddur** [comment]
Actually, WebP image may be converted from JPEG and from PNG. It seems to be difficult to determine if uploaded WebP image was converted from either JPEG/PNG.. 

Your proposal to convert WebP to fallback JPEG may be questionable, slow and/or far from optimal/comprehensive solution.

**Comment by com2** [comment]
Thank you for the good replies. With this I can prepare some well founded arguments for my recommendation. I think what you imply is that it is recommended to upload HQ jpeg/png media and let this plugin take care of generating the smaller but HQ webp derivatives, right?

**Reply by ddur** [reply]
> Thank you for the good replies. With this I can prepare some well founded arguments for my recommendation. I think what you imply is that it is recommended to upload HQ jpeg/png media and let this plugin take care of generating the smaller but HQ webp derivatives, right?

Right

**Reply by ddur** [reply]
> Thank you for the good replies. With this I can prepare some well founded arguments for my recommendation. I think what you imply is that it is recommended to upload HQ jpeg/png media and let this plugin take care of generating the smaller but HQ webp derivatives, right?

Default setting will do good job, but you can adjust compression quality and other settings to best fit your site type or speed requirements (in example: blog, (woo) commerce, HQ photo site ...)

**Comment by ddur** [comment]
@com2 

I got an idea how to do JPEG/PNG fallback for previously uploaded WebP images. :bulb:
But idea needs to be proven yet. :sunglasses:
